src/v/config/configuration.cc

```diff
@@ -4028,20 +4037,89 @@ configuration::configuration()
+  , iceberg_rest_catalog_aws_service_name(
+      *this,
+      "iceberg_rest_catalog_aws_service_name",
+      "AWS service name for SigV4 signing when using aws_sigv4 authentication "
+      "mode. Defaults to 'glue' for AWS Glue Data Catalog. Can be changed to "
+      "support other AWS services that provide Iceberg REST catalog APIs.",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::user},
+      "glue",
+      &validate_non_empty_string_opt)
+  , iceberg_rest_catalog_aws_access_key(
+      *this,
+      "iceberg_rest_catalog_aws_access_key",
+      "AWS access key for Iceberg REST catalog SigV4 authentication. If not "
+      "set, falls back to cloud_storage_access_key when using aws_sigv4 "
+      "authentication mode.",
+      {.needs_restart = needs_restart::yes,
+       .visibility = visibility::user,
+       .gets_restored = gets_restored::no},
+      std::nullopt,
+      &validate_non_empty_string_opt)
+  , iceberg_rest_catalog_aws_secret_key(
+      *this,
+      "iceberg_rest_catalog_aws_secret_key",
+      "AWS secret key for Iceberg REST catalog SigV4 authentication. If not "
+      "set, falls back to cloud_storage_secret_key when using aws_sigv4 "
+      "authentication mode.",
+      {.needs_restart = needs_restart::yes,
+       .visibility = visibility::user,
+       .secret = is_secret::yes,
+       .gets_restored = gets_restored::no},
+      std::nullopt,
+      &validate_non_empty_string_opt)
+  , iceberg_rest_catalog_aws_region(
+      *this,
+      "iceberg_rest_catalog_aws_region",
+      "AWS region for Iceberg REST catalog SigV4 authentication. If not set, "
+      "falls back to cloud_storage_region when using aws_sigv4 authentication "
+      "mode.",
+      {.needs_restart = needs_restart::yes,
+       .visibility = visibility::user,
+       .gets_restored = gets_restored::no},
+      std::nullopt,
+      &validate_non_empty_string_opt)
+  , iceberg_rest_catalog_aws_credentials_source(
+      *this,
+      "iceberg_rest_catalog_aws_credentials_source",
+      "Source of AWS credentials for Iceberg REST catalog SigV4 "
+      "authentication. "
+      "If not set, falls back to cloud_storage_credentials_source when using "
+      "aws_sigv4 authentication mode. Accepted values: config_file, "
+      "aws_instance_metadata, sts, gcp_instance_metadata, "
+      "azure_vm_instance_metadata, azure_aks_oidc_federation.",
+      {.needs_restart = needs_restart::yes,
+       .visibility = visibility::user,
+       .gets_restored = gets_restored::no},
+      std::nullopt,
+      {
+        model::cloud_credentials_source::config_file,
+        model::cloud_credentials_source::aws_instance_metadata,
+        model::cloud_credentials_source::sts,
+        model::cloud_credentials_source::gcp_instance_metadata,
+        model::cloud_credentials_source::azure_aks_oidc_federation,
+        model::cloud_credentials_source::azure_vm_instance_metadata,
+      })
```

Comment:
Can we mark these as gets_restored::yes (the default), and revert the changes to the ducktape test? The reason the original ones are marked as not getting restored is because they need to be set in the first place in order for a cluster restore to be run, which isn't the case here